dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -245,4 +252,43 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
             LOG_IF_FAILED(::CoRevokeClassObject(static_cast<DWORD>(token.Cookie())));
         }
     }
+
+    uint32_t PushNotificationManager::GetStringLengthFromComServer(const winrt::hstring& inputString)
+    {
+        try
+        {
+            wil::com_ptr<IWpnLrpPlatform> reunionEndpoint{
+                wil::CoCreateInstance<WpnLrpPlatform,
+                IWpnLrpPlatform>(CLSCTX_LOCAL_SERVER) };
+
+            std::wstring hstringAsCommonString{ inputString };
+            ULONG length = 0;
+            THROW_IF_FAILED(reunionEndpoint->GetStringLength(hstringAsCommonString.c_str(), &length));
+
+            return length;
+        }
+        catch (...)
+        {
+            HRESULT error = wil::ResultFromCaughtException();
+            return error;
+        }
+    }
+
+    uint32_t PushNotificationManager::ShutdownPlatformManually()
```

Comment:
Hmm why is WinSDK doing lifetime management for the LRP?

Reply:
It is just for testing purposes. The final version won't do this.